From adcad574478637c89b8c6947336d4a0287c19122 Mon Sep 17 00:00:00 2001
From: DigitalDreamtime <clive.messer@digitaldreamtime.co.uk>
Date: Fri, 6 May 2016 14:41:58 +0100
Subject: [PATCH 15/19] ES9023: hifiberry-dacpluslight-es9023-audio overlay

Add hifiberry-dacpluslight-es9023-audio overlay for HiFiBerry DAC+ Light.

##
## HiFiBerry DAC+ Light
##
##  card_name       = "snd_rpi_hifiberry_dac"
##  dai_name        = "HifiBerry DAC"
##  dai_stream_name = "HifiBerry DAC HiFi"
##
# dtoverlay=hifiberry-dacpluslight-es9023-audio
# dtoverlay=hifiberry-dacpluslight-es9023-audio,bclk_ratio_int_div
# dtoverlay=hifiberry-dacpluslight-es9023-audio,bclk_ratio_int_div,384k

Signed-off-by: DigitalDreamtime <clive.messer@digitaldreamtime.co.uk>

# Conflicts:
#	arch/arm/boot/dts/overlays/Makefile
---
 arch/arm/boot/dts/overlays/Makefile                |   1 +
 arch/arm/boot/dts/overlays/README                  |  20 ++++
 ...hifiberry-dacpluslight-es9023-audio-overlay.dts | 103 +++++++++++++++++++++
 3 files changed, 124 insertions(+)
 create mode 100644 arch/arm/boot/dts/overlays/hifiberry-dacpluslight-es9023-audio-overlay.dts

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index e8c1bd0..35b30ac 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -25,6 +25,7 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	hifiberry-amp.dtbo \
 	hifiberry-dac.dtbo \
 	hifiberry-dacplus.dtbo \
+	hifiberry-dacpluslight-es9023-audio.dtbo \
 	hifiberry-digi.dtbo \
 	hifiberry-digi-pro.dtbo \
 	hy28a.dtbo \
diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 2bc111a..f127148 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -457,6 +457,26 @@ Params: 24db_digital_gain       Allow gain to be applied via the PCM512x codec
                                 master for bit clock and frame clock.
 
 
+Name:   hifiberry-dacpluslight-es9023-audio
+Info:   Configures the HiFiBerry DAC+ Light audio card
+Load:   dtoverlay=hifiberry-dacpluslight-es9023-audio,<param>
+Params: 384k                    Instructs the es9023 codec driver to support
+                                352k8 and 384k sample rates.
+        bclk_ratio_int_div      Use bclk_ratio=50 for 16/24bps and
+                                bclk_ratio=100 for 32bps media when sample
+                                rate is a multiple of 8kHz and less than
+                                192kHz. Which causes the selection of the 19M2
+                                OSC as the parent for the PCM clock with an
+                                integer divider, rather than PLLD with
+                                fractional divider and MASH noise shaping.
+        card_name               Override the default, "snd_rpi_hifiberry_dac",
+                                card name.
+        dai_name                Override the default, "HifiBerry DAC",
+                                dai name.
+        dai_stream_name         Override the default, "HifiBerry DAC HiFi",
+                                dai stream name.
+
+
 Name:   hifiberry-digi
 Info:   Configures the HifiBerry Digi and Digi+ audio card
 Load:   dtoverlay=hifiberry-digi
diff --git a/arch/arm/boot/dts/overlays/hifiberry-dacpluslight-es9023-audio-overlay.dts b/arch/arm/boot/dts/overlays/hifiberry-dacpluslight-es9023-audio-overlay.dts
new file mode 100644
index 0000000..8fa451d
--- /dev/null
+++ b/arch/arm/boot/dts/overlays/hifiberry-dacpluslight-es9023-audio-overlay.dts
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2016 Clive Messer <clive.messer@digitaldreamtime.co.uk>
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Definitions for HiFiBerry DAC+ Light using ESS Sabre ES9023 DAC
+ *  eg. dtoverlay=hifiberry-dacpluslight-es9023-audio,384k,bclk_ratio_int_div
+ */
+
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2708";
+
+	fragment@0 {
+		target = <&i2s>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+
+	fragment@1 {
+		target-path = "/";
+		__overlay__ {
+			es9023_codec: es9023-codec {
+				#sound-dai-cells = <0>;
+				compatible = "ess,es9023";
+				status = "okay";
+			};
+		};
+	};
+
+	fragment@2 {
+		target = <&sound>;
+		es9023_dac: __overlay__ {
+			compatible = "hifiberry,es9023-dacpluslight";
+			/*
+			 * These are the default names that will be
+			 *  applied by the es9023-dac machine driver
+			 *  based on the compatible parameter.
+			 *
+			 * es9023-dac,card-name = "snd_rpi_hifiberry_dac";
+			 * es9023-dac,dai-name = "HifiBerry DAC";
+			 * es9023-dac,dai-stream-name = "HifiBerry DAC HiFi";
+			 */
+			status = "okay";
+			es9023-dac,cpu {
+				sound-dai = <&i2s>;
+			};
+			es9023-dac,codec {
+				sound-dai = <&es9023_codec>;
+			};
+		};
+	};
+
+	__overrides__ {
+		384k = <&es9023_codec>,"es9023,384k?";
+		bclk_ratio_int_div =
+			<&es9023_dac>,"es9023-dac,bclk-ratio-int-div?";
+		card_name = <&es9023_dac>,"es9023-dac,card-name";
+		dai_name = <&es9023_dac>,"es9023-dac,dai-name";
+		dai_stream_name = <&es9023_dac>,"es9023-dac,dai-stream-name";
+	};
+};
-- 
2.7.4

